In [10]:
import scipy.constants as sci
import numpy as np

Para obtener la rotación de la onda EM al pasar por el plasma podemos calcular la razón entre las componentes $x$ e $y$ del campo eléctrico $\vec{E}$, partiendo de la relación de Appleton-Hartree, llegando a (según *I.H. Hutchinson: Principles of Plasma Diagnostics*):
$$\frac{E_x}{E_y} = \frac{i Y \sin^2\theta}{2(1-X) \cos\theta} \pm i \left[ 1 + \frac{Y^2 \sin^4\theta}{4(1-X)^2 \cos^2 \theta} \right]^{1/2} $$  

Luego la rotación total que induce el plasma sobre la onda incidente sera:
$$ \alpha = \frac{\Delta \phi}{2} = \frac{e}{2 m_e c} \int_A^B{\frac{n_e \vec{B} \cdot \vec{dl}}{n_c (1 - n_e / n_c)^{1/2}}} $$

Donde la densidad crítica estaba dada por $n_c = \omega^2 m \epsilon_0 / e^2$. Y si consideramos $\frac{n_e}{n_c} \ll 1$, la expresión anterior se nos simplifica a:
$$ \alpha \approx \frac{e}{2 m_e c} \int_A^B{ \frac{n_e}{n_c} \vec{B} \cdot \vec{dl}} $$

Calculemos la constante que acompaña a la integral:

In [11]:
def acc_constant():
    fake_crit_density = sci.m_e * sci.epsilon_0 / sci.e ** 2 * (2 * sci.pi * sci.c) ** 2
    constant_out = sci.e / (2 * sci.m_e * sci.c)
    return constant_out / fake_crit_density

print(f"{acc_constant():.3e}")


2.631e-13


Entonces, usando este valor en la ecuación para $\alpha$, simplfiquemos el calculo del ángulo de rotación de Faraday: 
$$ \alpha \approx 2.631 \times 10^{-13} \cdot  \lambda^2 \int_A^B{n_e \vec{B} \cdot \vec{dl}} $$

In [12]:
def faraday_rotation(laser_wavelen, eon_density, mag_field, path_len):
    """
    laser_wavelen: float
    eon_density: float
    mag_field: array (3d vector)
    path_len: array (3d vector)
    """
    constant = acc_constant()
    angle_rotation = constant * laser_wavelen ** 2 * eon_density * np.dot(mag_field, path_len)
    return angle_rotation
    
green_wavelen = 532 * 10 ** -9
visible_spectrum = np.array([400 * 10 ** -9, 700 * 10 ** -9])
wavelens = np.linspace(visible_spectrum[0], visible_spectrum[1], 10 ** 5)
mag_field = np.array([0.0, 0.0, 1.0])
path_len = 2.0 * 10 ** -3
densities_range = np.array([10 ** 18, 10 ** 24])
eon_densities = np.linspace(densities_range[0], densities_range[1], 10 ** 5)
[faraday_rotation() for den in eon_densities ]
